extensions/amp-subscriptions/0.1/platform-store.js

```diff
@@ -44,6 +49,50 @@ export class EntitlementStore {
 
   }
 
+  /**
+   * Resolves a platform in the store
+   * @param {string} serviceId
+   * @param {!./subscription-platform.SubscriptionPlatform} platform
+   */
+  resolvePlatform(serviceId, platform) {
+    this.subscriptionPlatforms_[serviceId] = platform;
+  }
+
+  /**
+   * Returns the platform for the given id
+   * @param {string} servideId
+   * @returns {!./subscription-platform.SubscriptionPlatform}
+   */
+  getPlatform(servideId) {
+    const platform = this.subscriptionPlatforms_[servideId];
+    dev().assert(platform, `Platform for id ${servideId} is not resolved`);
+    return platform;
+  }
+
+  /**
+   * Returns the local platform;
+   * @returns {!./subscription-platform.SubscriptionPlatform}
+   */
+  getLocalPlatform() {
+    return this.getPlatform('local');
+  }
+
+  /**
+   * Returns all the platforms;
+   * @returns {!Array<!./subscription-platform.SubscriptionPlatform>}
+   */
+  getAllResolvedPlatforms() {
```

Comment:
Does it need to be public? The name is also confused - this is a registered platform, not resolved since "resolved" further used to mean "entitlements are known".